### PR TITLE
feat(contracts): add BurnMintWhitelistedTokenPool

### PR DESCRIPTION
**Pull request by roman-kashitsyn**

### CODE REVIEW

File: contracts/src/v0.8/ccip/pools/BurnMintWhitelistedTokenPool.sol [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+import {ITypeAndVersion} from "../../shared/interfaces/ITypeAndVersion.sol";
+import {IBurnMintWhitelisted} from "../interfaces/IBurnMintWhitelisted.sol";
+
+import {TokenPool} from "./TokenPool.sol";
+import {BurnMintTokenPoolAbstract} from "./BurnMintTokenPoolAbstract.sol";
+
+contract BurnMintWhitelistedTokenPool is BurnMintTokenPoolAbstract, ITypeAndVersion {
+  // solhint-disable-next-line chainlink-solidity/all-caps-constant-storage-variables
+  string public constant override typeAndVersion = "BurnMintTokenPool 1.2.0";
```
Comment (matYang) [inline contracts/src/v0.8/ccip/pools/BurnMintWhitelistedTokenPool.sol]: nit: version should be `1.4.0-dev`

File: core/gethwrappers/ccip/go_generate.go [hunk 2]
```diff
@@ -5,6 +5,7 @@ package ccip
 //go:generate go run ../generation/generate/wrap.go ../../../contracts/solc/v0.8.19/CommitStore/CommitStore.abi ../../../contracts/solc/v0.8.19/CommitStore/CommitStore.bin CommitStore commit_store
 //go:generate go run ../generation/generate/wrap.go ../../../contracts/solc/v0.8.19/CommitStoreHelper/CommitStoreHelper.abi ../../../contracts/solc/v0.8.19/CommitStoreHelper/CommitStoreHelper.bin CommitStoreHelper commit_store_helper
 //go:generate go run ../generation/generate/wrap.go ../../../contracts/solc/v0.8.19/BurnMintTokenPool/BurnMintTokenPool.abi ../../../contracts/solc/v0.8.19/BurnMintTokenPool/BurnMintTokenPool.bin BurnMintTokenPool burn_mint_token_pool
+//go:generate go run ../generation/generate/wrap.go ../../../contracts/solc/v0.8.19/BurnMintWhitelistedTokenPool/BurnMintWhitelistedTokenPool.abi ../../../contracts/solc/v0.8.19/BurnMintWhitelistedTokenPool/BurnMintWhitelistedTokenPool.bin BurnMintWhitelistedTokenPool burn_mint_whitelisted_token_pool
```
Comment (matYang) [inline core/gethwrappers/ccip/go_generate.go:8]: to add more confident, consider adding a test under `/test/pools/`, a good reference is `BurnFromMintTokenPoolSetup`

for this burn/mint signature, you'd need a mock token contract during testing; given it's a one-off, you can directly copy from actual customer contract for a test mock https://optimistic.etherscan.io/address/0x894134a25a5fac1c2c26f1d8fbf05111a3cb9487#code

File: contracts/src/v0.8/ccip/pools/BurnMintWhitelistedTokenPool.sol (hunk 1 above)
Comment (roman-kashitsyn) [reply contracts/src/v0.8/ccip/pools/BurnMintWhitelistedTokenPool.sol]: Good catch, fixed!

File: core/gethwrappers/ccip/go_generate.go (hunk 2 above)
Comment (roman-kashitsyn) [reply core/gethwrappers/ccip/go_generate.go:8]: I added a few tests, please take another look.

**Pull request closed without merging**